src/System.Net.Primitives/tests/FunctionalTests/IPAddressParsingSpan.cs

```diff
@@ -0,0 +1,494 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Net.Sockets;
+using Xunit;
+
+namespace System.Net.Primitives.Functional.Tests
+{
+    public class IPAddressParsingSpan
```

Comment:
It's great to see lots of tests added, but a lot of these look like duplication of existing tests, just for the span-based overloads instead of the array-based ones.  It'd be good to find a way to avoid the duplication and reuse tests once for both the array-based and span-based APIs.  There are a variety of approaches to this, but one example is here:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/RSA/EncryptDecrypt.cs
The tests are all in an abstract base class that exposes virtuals for each of the APIs that has an array vs span variant, and then derived types just override those methods, providing functionality to the base class to be tested.